lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -103,9 +104,15 @@ public LockImmutableTimestampResponse lockImmutableTimestamp() {
 
     @Override
     public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        return StartIdentifiedAtlasDbTransactionResponse.of(
-                lockImmutableTimestamp(),
-                TimestampAndPartition.of(getFreshTimestamp(), 0));
+        LockImmutableTimestampResponse immutableTimestamp = lockImmutableTimestamp();
+        try {
+            return StartIdentifiedAtlasDbTransactionResponse.of(
+                    immutableTimestamp,
+                    TimestampAndPartition.of(getFreshTimestamp(), 0));
+        } catch (Throwable throwable) {
+            unlock(ImmutableSet.of(immutableTimestamp.getLock()));
```

Comment:
Unlock should probably be wrapped in a try/catch, on `unlock` failure, the exception should be added as a suppressed exception to the first exception above, otherwise we lose the root cause :-)

Reply:
done